### DISCUSSION TITLE
Unable to set custom contextflyout template

### DISCUSSION THREAD

**Discussion by JankovicsSandor** [AvaloniaUI/Avalonia#6691] [Q&A] [answered]
Hi,
I would like to create a contextmenu for my treeview. I am using contextflyout due to having issues with the default contextmenu. For this I need to create menu items from a dynamic list.

I have tried this code 

```XML
<TreeView.ContextFlyout>
	<MenuFlyout Items="{Binding Path=ContextMenuItems,Mode=TwoWay}">
                <MenuFlyout.ItemTemplate>
                    <DataTemplate>
                          <MenuItem Header="{Binding Path=Header}"></MenuItem>
                    </DataTemplate>
                </MenuFlyout.ItemTemplate>
	</MenuFlyout>
</TreeView.ContextFlyout>
```
The desired result should be similar to this:
![image](https://user-images.githubusercontent.com/25934939/136524576-ec9a4e0a-9734-433d-851a-b6f92266872e.png)

The result is this
![image](https://user-images.githubusercontent.com/25934939/136523983-08f46f20-97d3-4e74-8cb7-b2f4c30a2ea4.png)

**Comment by workgroupengineering** [comment, marked as answer]
try
```xaml
<TreeView.ContextFlyout>
	<MenuFlyout Items="{Binding Path=ContextMenuItems,Mode=TwoWay}">
                <MenuFlyout.ItemTemplate>
                    <DataTemplate>
                          <TextBlock Text="{Binding Path=Header}"/>
                    </DataTemplate>
                </MenuFlyout.ItemTemplate>
	</MenuFlyout>
</TreeView.ContextFlyout>

```

**Reply by JankovicsSandor** [reply]
Thanks for the suggestion, already tried before. Still no change :(

**Reply by workgroupengineering** [reply]
> 
> 
> Thanks for the suggestion, already tried before. Still no change :(

Hi, I have tried it is actually there is a problem.
The MenuFlyout appears to ignore the local template.
To workaround this problem, move the template to Application.DataTemplates in App.axaml.

```xaml
    <Application.DataTemplates>
      <DataTemplate DataType="{x:Type vm:MenuViewModel}">
        <TextBlock Text="{Binding Path=Header}"/>
      </DataTemplate>
     
    </Application.DataTemplates>
```

**Reply by JankovicsSandor** [reply]
This way it works thanks.
